test/e2e/aks_tags.go

```diff
@@ -129,16 +130,14 @@ func AKSAdditionalTagsSpec(ctx context.Context, inputGetter func() AKSAdditional
 		}, inputGetter().WaitForUpdate...).Should(Succeed())
 		Eventually(checkTags, input.WaitForUpdate...).Should(Succeed())
 
-		if initialTags != nil {
-			By("Restoring initial tags for control plane")
-			expectedTags = initialTags
-			Eventually(func(g Gomega) {
-				g.Expect(mgmtClient.Get(ctx, client.ObjectKeyFromObject(infraControlPlane), infraControlPlane)).To(Succeed())
-				infraControlPlane.Spec.AdditionalTags = expectedTags
-				g.Expect(mgmtClient.Update(ctx, infraControlPlane)).To(Succeed())
-			}, inputGetter().WaitForUpdate...).Should(Succeed())
-			Eventually(checkTags, input.WaitForUpdate...).Should(Succeed())
-		}
+		By("Restoring initial tags for control plane")
+		expectedTags = initialTags
+		Eventually(func(g Gomega) {
+			g.Expect(mgmtClient.Get(ctx, client.ObjectKeyFromObject(infraControlPlane), infraControlPlane)).To(Succeed())
+			infraControlPlane.Spec.AdditionalTags = expectedTags
+			g.Expect(mgmtClient.Update(ctx, infraControlPlane)).To(Succeed())
+		}, inputGetter().WaitForUpdate...).Should(Succeed())
+		Eventually(checkTags, input.WaitForUpdate...).Should(Succeed())
```

Comment:
Looks like we're skipping the "delete all tags from the AzureManagedControlPlane resource" test case, because we now know if we do that we introduce test flakes?